src/address_family.rs

```diff
@@ -1,19 +1,31 @@
 use net2::UdpBuilder;
+#[cfg(not(windows))]
 use net2::unix::UnixUdpBuilderExt;
 use std::io;
 use std::net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, UdpSocket};
 use super::MDNS_PORT;
+#[cfg(not(windows))]
+use libc;
 
 pub enum Inet {}
 pub enum Inet6 {}
 
 pub trait AddressFamily {
     fn bind() -> io::Result<UdpSocket> {
         let addr = SocketAddr::new(Self::any_addr(), MDNS_PORT);
-        let socket = Self::socket_builder()?
-            .reuse_address(true)?
-            .reuse_port(true)?
-            .bind(&addr)?;
+        let builder = Self::socket_builder()?;
+        builder.reuse_address(true)?;
+	#[cfg(not(windows))]
+        match builder.reuse_port(true) {
+            Ok(_) => {},
+            // On linux kernel < 3.9 reuse_port is not available. Ignore
+            // the error on all linux kernel versions. If reuse_port is mandatory for some
+            // reason the bind() call will fail later.
+            #[cfg(target_os = "linux")]
+            Err(ref e) if e.raw_os_error() == Some(libc::ENOPROTOOPT) => {},
+            Err(err) => panic!("reuse_port failed: {}", err)
```

Comment:
Just return the error in an `Err`

Reply:
Thanks, will change this. I'm very new to rust...